### PR TITLE
fix: evaluate `Series` methods in the correct environment

### LINKED ISSUE #970: polars cannot identify the value in function
``` r
library(polars)
fn <- function() {
    message("I'm from function")
    fn_value <- pl$Series("a")
    pl$DataFrame(mtcars)$get_column("mpg")$cast(pl$String)$is_in(fn_value)
}
fn()
#> I'm from function
#> Error in pl$DataFrame(mtcars)$get_column("mpg")$cast(pl$String)$is_in(fn_value): object 'fn_value' not found
# "I'm out function"
fn_value <- pl$Series("a")
pl$DataFrame(mtcars)$get_column("mpg")$cast(pl$String)$is_in(fn_value)
#> polars Series: shape: (32,)
#> Series: 'mpg' [bool]
#> [
#>  false
#>  false
#>  false
#>  false
#>  false
#>  …
#>  false
#>  false
#>  false
#>  false
#>  false
#> ]
```

<sup>Created on 2024-03-27 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">

<summary>Session info</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.3.3 (2024-02-29)
#>  os       Ubuntu 22.04.4 LTS
#>  system   x86_64, linux-gnu
#>  ui       X11
#>  language en
#>  collate  C.UTF-8
#>  ctype    C.UTF-8
#>  tz       Asia/Shanghai
#>  date     2024-03-27
#>  pandoc   2.9.2.1 @ /usr/bin/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date (UTC) lib source
#>  cli           3.6.2   2023-12-11 [1] CRAN (R 4.3.3)
#>  digest        0.6.35  2024-03-11 [1] CRAN (R 4.3.3)
#>  evaluate      0.23    2023-11-01 [1] CRAN (R 4.3.3)
#>  fastmap       1.1.1   2023-02-24 [1] CRAN (R 4.3.3)
#>  fs            1.6.3   2023-07-20 [1] CRAN (R 4.3.3)
#>  glue          1.7.0   2024-01-09 [1] CRAN (R 4.3.3)
#>  htmltools     0.5.7   2023-11-03 [1] CRAN (R 4.3.3)
#>  knitr         1.45    2023-10-30 [1] CRAN (R 4.3.3)
#>  lifecycle     1.0.4   2023-11-07 [1] CRAN (R 4.3.3)
#>  magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.3.3)
#>  polars      * 0.15.1  2024-03-26 [1] Github (pola-rs/r-polars@c47431c)
#>  purrr         1.0.2   2023-08-10 [1] CRAN (R 4.3.3)
#>  R.cache       0.16.0  2022-07-21 [1] CRAN (R 4.3.3)
#>  R.methodsS3   1.8.2   2022-06-13 [1] CRAN (R 4.3.3)
#>  R.oo          1.26.0  2024-01-24 [1] CRAN (R 4.3.3)
#>  R.utils       2.12.3  2023-11-18 [1] CRAN (R 4.3.3)
#>  reprex        2.1.0   2024-01-11 [1] CRAN (R 4.3.3)
#>  rlang         1.1.3   2024-01-10 [1] CRAN (R 4.3.3)
#>  rmarkdown     2.26    2024-03-05 [1] CRAN (R 4.3.3)
#>  sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.3.3)
#>  styler        1.10.2  2023-08-29 [1] CRAN (R 4.3.3)
#>  vctrs         0.6.5   2023-12-01 [1] CRAN (R 4.3.3)
#>  withr         3.0.0   2024-01-16 [1] CRAN (R 4.3.3)
#>  xfun          0.42    2024-02-08 [1] CRAN (R 4.3.3)
#>  yaml          2.3.8   2023-12-11 [1] CRAN (R 4.3.3)
#> 
#>  [1] /home/yun/Rlibrary/4.3
#>  [2] /usr/local/lib/R/site-library
#>  [3] /usr/lib/R/site-library
#>  [4] /usr/lib/R/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

### PR DESCRIPTION
**Pull request by etiennebacher**
Close #970

### CODE REVIEW

File: tests/testthat/test-series-sub-namespace.R [hunk 1]
```diff
@@ -50,3 +50,38 @@ test_that("categorical sub namespace", {
...
+    fn_value = pl$Series(values = "a")
+    as.vector(pl$Series(values = letters[1:3])$is_in(fn_value))
+  }
+
+  expect_identical(
+    fn(),
+    c(TRUE, FALSE, FALSE)
+  )
+
+  # Input is passed by the suer
```
Comment (george-wood) [inline tests/testthat/test-series-sub-namespace.R:67]: Small typo here: "suer" -> "user"

**Comment by eitsupi** [comment]
I was researching this as well. Please see this SO answer.
https://stackoverflow.com/questions/55813859/non-standard-evaluation-in-a-user-defined-function-with-lapply-or-with-in-r

As an important correction, we need to use `match.call` here, not `sys.call`.

``` r
f <- function(x, ..., y = 0) {x + y}

g1 <- function(x, y) {do.call(f, as.list(match.call()[-1]), envir = parent.frame())}
h1 <- function() {a <- 100; g1(1, a)}

g2 <- function(x, y) {do.call(f, as.list(sys.call()[-1]), envir = parent.frame())}
h2 <- function() {a <- 100; g2(1, a)}

h1()
#> [1] 101
h2()
#> [1] 1
```

<sup>Created on 2024-03-27 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by etiennebacher** [comment]
Seems like we use `sys.call()` in several other places, should those be replaced too? 

Otherwise, I think this is good to go, failures are due to some CI stuff

**Comment by eitsupi** [comment]
> Seems like we use `sys.call()` in several other places, should those be replaced too?

I looked at it and it probably does not need to be updated.

I am thinking about tests that should be added. Once the tests are added, I merge this.

File: R/series__series.R [hunk 2]
```diff
@@ -159,6 +159,8 @@ Series_shape = method_as_active_binding(\() .pr$Series$shape(self))
 add_expr_methods_to_series = function() {
   methods_exclude = c(
     "agg_groups",
+    "over",
+    "rolling",
```
Comment (eitsupi) [inline R/series__series.R:163]: I have removed these as they are not directly related to this PR, but obviously they do not work on the Series.

**Review by eitsupi — APPROVED** [review]
CI failure is unrelated to this change.

**Pull request merged**